src/fork_gc.c

```diff
@@ -752,15 +761,31 @@ static int ForkGc_PeriodicCallback(RedisModuleCtx *ctx, void *privdata) {
   int ret_val = 1;
 
   size_t totalCollectedBefore = gc->stats.totalCollected;
+  pid_t ppid_before_fork = getpid();
 
   TimeSampler_Start(&ts);
   pipe(gc->pipefd);  // create the pipe
   RedisModule_ThreadSafeContextLock(ctx);
-  cpid = fork();  // duplicate the current process
+  cpid = ForkGc_Fork();  // duplicate the current process
   RedisModule_ThreadSafeContextUnlock(ctx);
+
+  if (cpid == -1) {
+    return 1;
+  }
+
   if (cpid == 0) {
     // fork process
     close(gc->pipefd[GC_READERFD]);
+    if (!ForkGc_IsForkApiExists()) {
+      // set the parrent death signal to SIGTERM
+      int r = prctl(PR_SET_PDEATHSIG, SIGTERM);
```

Comment:
This breaks builds on non-Linux systems (i.e. OS X)

Reply:
I know, we need to use some ifdef, but we need to fast so I think we can add this later on master ..